_posts/2017-12-27-cheap-docker-swarm-cluster-aws-terraform.md

```diff
@@ -0,0 +1,98 @@
+
+But we also need to consider the cost of electricity.  Let’s assume a $0.10/kWh rate, which is quite inexpensive, but also common across large parts of North America.  Assuming a 300W constant power draw, we are looking at $21.60/month, just for electricity.  Our cost of electricity is about the same amount as our amortized hardware costs!  Not really what I had in mind for cheap, but this is just our baseline.
+
+Ignoring a large number of other considerable factors, this gives us an estimated monthly cost of operation of $42.43.  This is the amortized cost of the hardware, plus the cost of electricity.
+
+There are a number of alternatives we could explore on the self hosted side that could reduce our costs.  For example, we could explore the use of low power ARM processors, which are both inexpensive as well as efficient.  Depending on the intended purpose, this may be a valid option.  But for the purpose of comparison, we will stick with an Intel processor.
+
+To summarize:
+
+| Description | Cost |
+| ----------- | ---- |
+| Monthly Amortized Hardware Costs | $20.83 |
+| Monthly Electricity Usage        | $21.60 |
+| Total                            | $42.43 |
+
+## Cloud Hosting
+
+With [AWS](https://aws.amazon.com/), there are a large number of server types to choose from.  A comparable EC2 type to our self-hosted solution would be the t2.medium instance.  [Here’s a great site that compares the various options and prices.](https://www.ec2instances.info/)  If we use on-demand pricing, we will get higher availability, but we are paying full price.  We have the option to go with spot pricing and we are going to sacrifice availability for some savings.  Here’s the comparison for the t2.medium, with current 2017 pricing.
+
+
+| Lifecycle | Total Hourly Cost | Total Monthly Cost |
+| --------- | ----------------- | ------------ |
+| On-Demand | $0.0464 | $33.41 |
+| Spot | $0.0185 | $13.32 |
+
+Even with full availability, we are still getting cheaper prices than our self-build, but a significant savings when we look at spot pricing.
+
+## Setting it all up
+
+Another major factor we need to discuss is how long it takes to set this stuff up.  We’ve done a simple comparison which makes a compelling case for the cloud.  But where the cloud really shines is in the time savings setting the cluster up.
+
+It look me about 2 hours to build each of my self-hosted boxes, that’s after sourcing all of the parts.  Let’s say a total of 6 hours, assuming everything goes well.
```

Comment:
I bought a server from Wal-mart for about what you priced your servers - wait, there's more than one? Are they $500 each or $500 total? If it's total, cool! You didn't say that earlier. If it's not, well, why build when you can buy?